File: backend-dotnet/SahayaSetu/Services/IAdminService.cs
using SahayaSetu.DTO;

namespace SahayaSetu.Services
{
    public interface IAdminService
    {
        Task<List<NgoResponseDto>> GetAllNgosAsync();
        Task<List<DonorResponseDto>> GetAllDonorsAsync();
        Task<List<RequestResponseDto>> GetAllRequestsAsync();
        Task ApproveNgoAsync(int ngoId);
        Task DisapproveNgoAsync(int ngoId); // Reject
        Task DisableNgoAsync(int ngoId);
        Task EnableNgoAsync(int ngoId);
    }
}

File: backend-java/sahayaSetu/src/main/java/com/sahayaSetu/services/IAdminService.java
package com.sahayaSetu.services;

import com.sahayaSetu.dtos.NgoResponseDto;
import com.sahayaSetu.dtos.RequestResponseDto;
import com.sahayaSetu.dtos.DonorResponseDto;

import java.util.List;

public interface IAdminService {
    List<NgoResponseDto> getAllNgos();
    List<RequestResponseDto> getAllRequests();
    List<DonorResponseDto> getAllDonors();
    void approveNgo(Long ngoId);
    void disapproveNgo(Long ngoId);
    void disableNgo(Long ngoId);
    void enableNgo(Long ngoId);
}
